apps/email/js/cards/compose.js

```diff
@@ -382,6 +406,9 @@ return [
     _saveNeeded: function() {
       var self = this;
       var checkAddressEmpty = function() {
```

Comment:
It feels like the new checkAddressesPassed() function (or whatever the new name might be) serves the purpose of checkAddressEmpty(): we just want to know if an address changed, and if so save. In the case where the start state is no addresses then if none still, then that qualifies as no change. However, if the user started with addresses and removed them all that would be a save worth prompting.

So I would just call this.checkAddressEmpty() directly and remove this internal checkAddressEmpty function.